src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,151 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private Node<T> prev;
+        private T data;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T data, Node<T> next) {
+            this.prev = prev;
+            this.data = data;
+            this.next = next;
+        }
+    }
+
     @Override
-    public void add(T value) {
+    public void add(T data) {
+        Node<T> newNode;
+        if (head == null) {
+            newNode = new Node<>(null, data, null);
+            head = tail = newNode;
+        } else {
+            newNode = new Node<>(this.tail, data, null);
+            newNode.prev = tail;
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
-    public void add(T value, int index) {
+    public void add(T data, int index) {
+        checkIndex(index, size);
+        Node<T> newNode;
+        if (head == null) {
+            newNode = new Node<>(null, data, null);
+            head = tail = newNode;
+        } else if (index == 0) {
+            newNode = new Node<>(null, data, head);
+            head = newNode;
+            newNode.next.prev = newNode;
+        } else if (index == size) {
+            add(data);
+            return;
```

Comment:
You can combine first and last condition
`if (head == null || index == size) {
add(data);
}`